### ISSUE TITLE
 Four Android emulators, two apps

### ISSUE THREAD

**Issue by bitsgalore** [bitsgalore/bitsgalore.github.io#74]
<https://www.bitsgalore.org/2021/02/09/four-android-emulators-two-apps>

**Comment by euanc** [comment]
Great post thank you Johan!

I thought I'd add a response to Pennock, May and Day's comment that you quoted at the end:

 _"we must find some way to deal with apps which have an inherent reliance on content hosted externally to the app."_

The Emulation as a Service team at the University of Freiburg have started addressing this and gave a [great presentation](https://osf.io/2cqne/) in collaboration with folks from the University of Amsterdam about this kind of thing at iPRES 2019 ([direct download](https://osf.io/2cqne/download)). While not directly addressing mobile apps/contexts the general approach of preserving and/or simulating obsolete web services and connecting them in emulation should be applicable to this use-case.

**Comment by gewappnet** [comment]
It would be interesting to explore the possibilities to integrate one of the tested emulators (preferably Android Emulator) in the Emulation as a Service framework. But the other challenge would be how a collection of Android apps would look like. I think currently no cultural heritage organizations is collecting mobile apps. So with the possibility of giving access to these app via emulation, we could develop a collection workflow for these kind of apps. What do we have to collect (file formats, metadata)?

For iOS apps it might be interesting that the new Macs are able to run iOS apps. Maybe this could lead to a working solution for preservation.

**Comment by euanc** [comment]
I believe the National Institute for Standards and Technology has been collecting apps in the National Software Reference Library ([NSRL](https://www.nist.gov/itl/ssd/software-quality-group/national-software-reference-library-nsrl)) and we may be able to access them in EaaSI. You can find lists of the Apps they have calculated hashes for [here](https://www.nist.gov/itl/ssd/software-quality-group/national-software-reference-library-nsrl/nsrl-download/current-rds). 

As part of the second round of [EaaSI](https://www.softwarepreservationnetwork.org/emulation-as-a-service-infrastructure/) funding from the [Sloan](https://sloan.org/) and [Mellon](https://mellon.org/) foundations we committed to adding an Android emulator to EaaSI so this is on our agenda for the next 18 months, but currently is scheduled towards the end of that period. 

Workflows are a challenge and question as part of that work, so any ideas related to that would be much appreciated. The [EaaSI forum](http://forum.eaasi.cloud/) might be a good place to discuss those also, it's open to all! (Johan [posted](https://forum.eaasi.cloud/t/emulation-of-mobile-devices-esp-android/147/8) about this topic there also, thank you Johan!)

**Comment by bitsgalore** [comment]
@euanc Yes I remember seeing that presentation at iPres. Scalability will be an issue though, re-creating a complete server infrastructure for one app is one thing, but doing that for larger collections of apps is something else. Might be interesting to explore as a proof-of-concept for one or two selected apps. (BTW this is in response to your first comment; seems your second comment was posted at the exact same time as I was writing this one.)

**Comment by bitsgalore** [comment]
@gewappnet I'm about to start the next phase of the mobile apps work, which will cover collection workflows. This will  include how to download the installers outside the app store, and (tools for extracting) technical metadata. Once that's done I'll report the results in another blog post. I'll also try to include iOS apps in this follow-up (even though AFAIK no good emulation options exist for that platform yet).

**Comment by euanc** [comment]
@bitsgalore I'm excited to here how all that goes, we'll definitely be watching out for it to inform our work in EaaSI.

In regards to iOS: [Corellium](https://corellium.com/) are doing iOS _virtualization_ (emulation may take quite a bit longer as apple chips are bleeding edge and emulation always lags) and might be worth speaking with. They recently had a pretty good result in their [court case with Apple](https://arstechnica.com/tech-policy/2020/12/corellium-notches-partial-victory-in-apple-ios-copyright-case/).

**Comment by bitsgalore** [comment]
@euanc Yes, my colleague @samalloing already brought Corellium to my attention. From their website I got the impression they were only offering a cloud based service, but the Arstechnica piece suggests otherwise? Anyway, it's on my list of potential solutions to check out at some time.

**Comment by euanc** [comment]
I'd recommend reaching out to them. I've started a conversation in the past and they were receptive. Their technology is quite flexible also.

**Comment by bitsgalore** [comment]
I just came across the new [Waydroid](https://github.com/waydroid/waydroid) project, which [according to this piece](https://www.xda-developers.com/waydroid-android-apps-on-linux/) is:

>  a rebuilt version of Anbox that is designed to use more of the host device’s native hardware than Anbox — which means faster performance. The project’s main focus is running Android apps on Halium-based Linux phones (Halium is similar in concept to Android’s GSI, but for standard Linux), but it can also run on any devices with a mainline Linux kernel.

**Comment by upintheairsheep** [comment]
iOS emulation projects are actually starting now: https://emulation.gametechwiki.com/index.php/IOS_emulators
Also see: https://emulation.gametechwiki.com/index.php/Cellphone_emulators
